web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -99,11 +102,15 @@ export function ChooseCreditsForm({
 
   const [spendingCap, setSpendingCap] = useState(0);
   const [creditsAvailable, setCreditsAvailable] = useState(0);
+  // A big number here avoids the NOT_ENOUGH_BALANCE error on rendering
+  // the form until the actual user balance is fetched
+  const [userBalance, setUserBalance] = useState(10101010101010101);
```

Comment:
this is a bit hacky, what about displaying a loading spinner instead of the form while we fetch the balance?

Reply:
Since the data we are fetching is related to validation for an error users may never encounter, I thought it'd be better to not block the UI from rendering, so users can interact with the form straight away. But yes, a loading animation is a more conventional solution.